cpg_infra/driver.py

```diff
@@ -1187,26 +1191,33 @@ def main_tmp_bucket(self):
             'main-tmp',
             lifecycle_rules=[self.infra.bucket_rule_temporary()],
             versioning=False,
+            autoclass=False,  # Gets cleared out automatically.
         )
 
     @cached_property
     def main_analysis_bucket(self):
         return self.infra.create_bucket(
-            'main-analysis', lifecycle_rules=[self.infra.bucket_rule_undelete()]
+            'main-analysis',
+            lifecycle_rules=[self.infra.bucket_rule_undelete()],
+            autoclass=self.dataset_config.autoclass,
         )
 
     @cached_property
     def main_web_bucket(self):
         return self.infra.create_bucket(
-            'main-web', lifecycle_rules=[self.infra.bucket_rule_undelete()]
+            'main-web',
+            lifecycle_rules=[self.infra.bucket_rule_undelete()],
+            autoclass=self.dataset_config.autoclass,
         )
 
     @cached_property
     def main_upload_buckets(self) -> dict[str, Any]:
         main_upload_undelete = self.infra.bucket_rule_undelete(days=30)
         main_upload_buckets = {
             'main-upload': self.infra.create_bucket(
-                'main-upload', lifecycle_rules=[main_upload_undelete]
+                'main-upload',
+                lifecycle_rules=[main_upload_undelete],
+                autoclass=self.dataset_config.autoclass,
```

Comment:
Do we want things in upload to move to archive? I think things in main-upload should start throwing warnings if they haven't been actioned in a reasonable amount of time.

Reply:
Well, it doesn't hurt to have Autoclass enabled. There certainly have been files in upload for some period of time, but I agree that ideally we'd also have an alert. That's a bit orthogonal though.